linux/keyman-config/keyman_config/ibus_util.py

```diff
@@ -171,21 +169,27 @@ def _start_ibus_daemon(realuser):
 
         if realuser:
             # we have been called with `sudo`. Start ibus-daemon for the real user.
-            logging.info('starting ibus-daemon for user %s', realuser)
-            subprocess.run(['sudo', '-u', realuser].extend(args), check=True)
+            args = ['sudo', '-u', realuser,
+                    f'DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/{os.environ.get("SUDO_UID")}/bus',
+                    f'XDG_RUNTIME_DIR=/run/user/{os.environ.get("SUDO_UID")}', *args]
+            logging.info(f'starting ibus-daemon for user {realuser} with args: {args}')
         else:
             logging.info('ibus-daemon not running. Starting it...')
-            subprocess.run(args, check=True)
-    except Exception:
-        logging.warning('Failed to start ibus-daemon')
+            logging.debug(f'Starting ibus-daemon with args: {args}')
```

Comment:
Do we need this as well as l.177? Should we just combine the args data into l.177 like we have with l.175?

Reply:
done